lib/pxutil.js

```diff
@@ -138,17 +138,56 @@ function filterConfig(config) {
 
 /**
  * extractCookieNames - Extract all the cookie names that were sent in the cookie http header.
- * @param {Object} cookieHeader - The received http request cookie header.
+ * @param {Array} cookieHeader - The received http request cookie header after parsing by delimiter.
  */
-function extractCookieNames(cookieHeader) {
+function extractCookieNames(cookies) {
+    return cookies.map(cookie => cookie.split('=')[0].trim());
+}
+
+/**
+ *
+ * */
+function extractCookieValue(cookieName, cookieArray) {
     let result;
-    if(cookieHeader) {
-        const cookies = cookieHeader.split(';');
-        result = cookies.map(cookie => cookie.split('=')[0].trim());
+    for (let i = 0 ; i < cookieArray.length ; i++) {
+        const keyValue = cookieArray[i].split('=');
+        if (keyValue[0] === cookieName) {
+            result = keyValue[1];
+            break;
+        }
     }
     return result;
 }
 
+/**
+ * prepareCustomParams - if there's a enrich custom params handler configured on startup,
+ * it will populate to @dict with the proper custom params
+ * @param {pxconfig} config - The config object of the application
+ * @param {object} dict - the object that should be populated with the custom params
+ * */
+function prepareCustomParams(config, dict) {
+    const customParams = {
+        'custom_param1': '',
+        'custom_param2': '',
+        'custom_param3': '',
+        'custom_param4': '',
+        'custom_param5': '',
+        'custom_param6': '',
+        'custom_param7': '',
+        'custom_param8': '',
+        'custom_param9': '',
+        'custom_param10': ''
+    };
+    if (config.ENRICH_CUSTOM_PARAMETERS) {
+        const riskCustomParams = config.ENRICH_CUSTOM_PARAMETERS(customParams);
```

Comment:
`riskCustomParams` no longer seems like a good name for this, as it's not just risk anymore